hazelcast/src/main/java/com/hazelcast/client/impl/connection/nio/ClientConnectionManagerImpl.java

```diff
@@ -790,7 +790,9 @@ private void authenticateOnCluster(ClientConnection connection) {
                 handleSuccessfulAuth(connection, response);
                 break;
             case CREDENTIALS_FAILED:
-                AuthenticationException authException = new AuthenticationException("Invalid credentials!");
+                AuthenticationException authException = new AuthenticationException("Invalid credentials!"
+                        + " Either ClientConfig.getClusterName() or"
+                        + " ClientConfig.getSecurityConfig() is inconsistent with the cluster config.");
```

Comment:
how about ? "Authentication failed. The configured cluster name on the client (see ClientConfig.setClusterName()) does not match the one configured in the cluster or the credentials set in the Client security config could not be authenticated."